aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -3489,6 +3489,8 @@ Tensor _weight_int4pack_mm_cpu(
   TORCH_CHECK(qGroupSize == 32 || qGroupSize == 64 || qGroupSize == 128
       || qGroupSize == 256,
       __func__, ": expect qGroupSize to be 32, 64, 128 or 256, got ", qGroupSize);
+  TORCH_CHECK(K % qGroupSize == 0,
```

Comment:
Hi, something unrelated to your implementation, but related to this function - would it be okay to add another assert that checks whether the value of groupsize is as expected (based on the shape of weights)? Thanks!

Reply:
What does it mean by `as expected`?

Reply:
the qgroupsize should be equal to `(B.numel() * 2)/(qScalesAndZeros.numel() / 2)`

Reply:
I see.

Reply:
Since it's irrelevant to this PR, we may add the check if an issue is reported in the future.

Reply:
OK, thanks!